app/lib/actions/remote_execution/event_helpers.rb

```diff
@@ -0,0 +1,41 @@
+module Actions
+  module RemoteExecution
+    module EventHelpers
+      module ClassEventHelpers
+        def event_states
+          []
+        end
+
+        def event_names
+          event_states.map do |state|
+            event_name_base + '_' + event_name_suffix(state).to_s
+          end
+        end
+
+        def feature_job_event_names(label)
+          event_states.map do |state|
+            ::Foreman::Observable.event_name_for("#{event_name_base}_#{label}_#{event_name_suffix(state)}")
+          end
+        end
+      end
+
+      def self.included(base)
+        # Added by default by ObservableAction
+        base.execution_plan_hooks.do_not_use :emit_event, :on => :success
+        base.extend ClassEventHelpers
+      end
+
+      def emit_event(execution_plan, hook)
+        return unless root_action?
+
+        if input["job_features"]&.any?
+          payload = event_payload(execution_plan)
+          input['job_features'].each do |feature|
+            name = "#{self.class.event_name_base}_#{feature}_#{self.class.event_name_suffix(hook)}"
+            trigger_hook name, payload: payload
+          end
```

Comment:
The following change sends 'Host(s) Job' event, if no features are specified.
We could also always trigger that event since :thinking: 
```suggestion

        payload = event_payload(execution_plan)
        if input["job_features"]&.any?
          input['job_features'].each do |feature|
            name = "#{self.class.event_name_base}_#{feature}_#{self.class.event_name_suffix(hook)}"
            trigger_hook name, payload: payload
          end
        else
          name = "#{self.class.event_name_base}_#{self.class.event_name_suffix(hook)}"
          trigger_hook name, payload: payload
        end
```